src/web/app/user-join-page.component.ts

```diff
@@ -49,30 +49,28 @@ export class UserJoinPageComponent implements OnInit {
         return;
       }
 
-      this.courseService.getJoinCourseStatus(this.key, this.entityType).subscribe((resp: JoinStatus) => {
-        this.hasJoined = resp.hasJoined;
-        this.userId = resp.userId || '';
-        if (this.hasJoined && this.userId) {
-          // The regkey has been used and there is a logged in user.
-          // Simply redirect the user to their home page, regardless of whether the regkey matches or not.
-          this.navigationService.navigateByURL(this.router, `/web/${this.entityType}/home`);
-        } else {
+      const nextUrl: string = `${window.location.pathname}${window.location.search.replace(/&/g, '%26')}`;
+      this.authService.getAuthUser(undefined, nextUrl).subscribe((auth: AuthInfo) => {
+        if (!auth.user) {
           this.isLoading = false;
+          window.location.href = `${this.backendUrl}${auth.studentLoginUrl}`;
```

Comment:
This is not 100% accurate. Depending on the entity type (instructor or student), the login URL should be different.